<?xml version="1.0"?>
<launch>
  
    <group ns="axis_camera">		
		<!-- Flag to enable/disable the execution of image republish and proc -->
		<arg name="run_republish" value="true"/>
		<arg name="ptz_enabled" value="true"/>
		
		<param name="hostname" value="192.168.0.185" />
		<!--param name="hostname" value="192.168.0.73" /-->
		<param name="username" value="root" />
		<param name="password" value="R0b0tn1K" />
		<param name="camera_number" value="1" />
		<!--param name="camera_model" value="axis_p5512"/-->
		<param name="camera_model" value="axis_p5534"/>
		<param name="camera_id" value="XXXX"/>
		<param name="fps" value="0"/>
		<param name="compression" value="0"/>
		<param name="frame" value="axis_camera1"/>
		<param name="profile" value="TestQuality"/>
		<param name="enable_auth" value="false"/>
		
		<rosparam file="$(find axis_camera)/config/ptz_config.yaml" command="load" if="$(arg ptz_enabled)"/>
		
		<!-- Important: Every model has its own config file. This config file will depend on the Axis configuration -->
		<param name="camera_info_url" value="package://axis_camera/data/axis_p5534.yaml"/>
		
		<node pkg="axis_camera" type="axis_node.py" name="axis_camera" output="screen"/>
		<!-- Republishes from Compressed to raw -->
		<node pkg="image_transport" type="republish" name="axis_republisher" output="screen" args="compressed in:= raw out:=image_raw" if="$(arg run_republish)"/>
		<!-- IMAGE PROC: Republish image topics by using camera info -->
		<node pkg="image_proc" type="image_proc" name="axis_image_proc" output="screen" if="$(arg run_republish)"/>
	
	</group>
	

</launch>
